homeassistant/components/cover/somfy_mylink.py

```diff
@@ -0,0 +1,206 @@
+"""
+Platform for the Somfy MyLink device supporting the Synergy JsonRPC API.
+
+For more details about this platform, please refer to the documentation
+https://home-assistant.io/components/cover.somfy_mylink/
+"""
+import logging
+
+from homeassistant.components.cover import (
+    ATTR_POSITION, SUPPORT_CLOSE, SUPPORT_OPEN, SUPPORT_SET_POSITION,
+    SUPPORT_STOP, CoverDevice)
+from homeassistant.components.somfy_mylink import (
+    CONF_COVER_OPTIONS, DATA_SOMFY_MYLINK)
+from homeassistant.helpers.event import async_track_time_change
+
+_LOGGER = logging.getLogger(__name__)
+DEPENDENCIES = ['somfy_mylink']
+DEFAULT_SUPPORTED_FEATURES = (SUPPORT_OPEN | SUPPORT_STOP | SUPPORT_CLOSE)
+
+
+async def async_setup_platform(hass,
+                               config,
+                               async_add_entities,
+                               discovery_info=None):
+    """Discover and configure Somfy covers."""
+    somfy_mylink = hass.data[DATA_SOMFY_MYLINK]
+    cover_options = discovery_info.get(CONF_COVER_OPTIONS, [])
+    cover_list = list()
+    mylink_status = await somfy_mylink.status_info()
+    for cover in mylink_status['result']:
+        cover_config = dict()
+        cover_config['target_id'] = cover['targetID']
+        for cover_opt in cover_options:
+            if cover_opt.get('name') in ('*', cover['name']):
```

Comment:
Star is a wildcard. Common use case is where a user has multiple awnings which commonly require reversal of the up/down commands.

Rather than explicitly defining every awning (and needing to update `configuration.yaml` when they change) they can use a wildcard entry to change the default behaviour (I.E `reverse: true`) and then explicitly define the exceptions if required.

I agree it's not a great way to do it but I looked lots of components trying to see how that sort of functionality had been implemented and couldn't find any examples of it. Only other alternative I could think of was to have new config constants like `default_reverse` and `default_move_time` but that is also pretty messy!